### PR TITLE
clang-format: Place opening braces on starting context

### PR DESCRIPTION
**Pull request by danobi**
This PR does a few things. I am happy to split it out into multiple PRs if necessary.

1. Simplify and make more reliable clang-format GHA job
2. Remove unused .clang-formoat config
3. Codemod the entire codebase to place opening braces on starting context

Note this PR does not yet update `.git-blame-ignore-revs`. That will come in a
separate PR cuz when this is merged the SHAs will be different. See
https://michaelheap.com/git-ignore-rev/ for more details on how it will work.

##### Checklist

- [ ] Language changes are updated in `man/adoc/bpftrace.adoc` and if needed in `docs/reference_guide.md`
- [ ] User-visible and non-trivial changes updated in `CHANGELOG.md`
- [ ] The new behaviour is covered by tests

### CODE REVIEW

**Comment by danobi** [comment]
Unfortunately this PR, when merged, will probably conflict out every other PR. But hopefully it should be fairly straightforward to resolve the moving of braces.

**Review by viktormalik — COMMENTED** [review]
I like the way kernel does this - it places the brace on a new line for function defs and on the context line everywhere else. It makes (especially small) functions nicely visually distinguishable in the source.

Would it be possible to achieve something similar with clang-format?

**Comment by danobi** [comment]
Yep - done

File: .clang-format [hunk 1]
```diff
@@ -20,25 +20,10 @@ AlwaysBreakBeforeMultilineStrings: false
...
-  BeforeCatch:           true
-  BeforeElse:            true
-  IndentBraces:          false
-  SplitEmptyFunction:    true
-  SplitEmptyRecord:      true
-  SplitEmptyNamespace:   true
 BreakBeforeBinaryOperators: None
 BreakBeforeBraces: Custom
+BraceWrapping:
+  AfterFunction: true
```
Comment (viktormalik) [inline .clang-format:26]: Nit: why moving this out of alphabetic order?

**Review by viktormalik — APPROVED** [review]
LGTM (obviously I didn't check the formatting commit).

**Comment by jordalgo** [comment]
Love it!

**Review by jordalgo — APPROVED** [review]

File: scripts/seccomp.c [hunk 2]
```diff
@@ -151,35 +156,35 @@ int main(int argc, char** argv) {
...
+              ctx, SCMP_ACT_KILL, SCMP_SYS(bpf), 1, SCMP_A0(SCMP_CMP_EQ, v));
+          if (rc < 0) {
+            printf("Failed to add KILL filter for command: %s: %s\n",
+                   optarg,
+                   strerror(-1 * rc));
+          } else {
+            printf("Added KILL for command: %s\n", optarg);
+          }
         } else {
-          printf("Added KILL for command: %s\n", optarg);
```
Comment (jordalgo) [inline scripts/seccomp.c:170]: drive by fix?

File: .clang-format (hunk 1 above)
Comment (danobi) [reply .clang-format:26]: Ah, didn't realize it was ordered. I wanted to move it closer to the `Custom` bit that enables `BraceWrapping`

Comment (viktormalik) [reply .clang-format:26]: No problem with that, I was just wondering why the move. Feel free to leave it as it is.

File: scripts/seccomp.c (hunk 2 above)
Comment (jordalgo) [reply scripts/seccomp.c:170]: Ah I see it now. Haha

**Pull request merged**